### PR TITLE
`Learning path`: Improve generation of missing learning path

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

Get requests should not create new persisted data. Although this would happen for the getLearningPathId endpoint in very rare cases, we should try our best to keep the API REST conform.

### Description

Refactors the endpoint to get the learning path id to return an error if the learning path does not exist and adds a new endpoint to create one.

### Steps for Testing

Prerequisites:
- 1 Students
- 1 Course with learning paths enabled

1. Log in to Artemis
2. Navigate to Course > Learning Path
3. There should be no error
4. Navigate back to another view, e.g. Course Overview
5. Delete the Learning Path from the Database (drop the row with the corresponding id in `learning_path`)
6. Navigate to Course > Learning Path
7. There should be no error

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **New Features**
	- Added functionality to generate a learning path for a specific course.
- **Enhancements**
	- Improved error handling for learning paths in both backend and frontend components.
- **Refactor**
	- Refactored existing logic for better clarity and efficiency in learning path generation and testing.
- **Tests**
	- Expanded integration and component tests to cover new scenarios and error handling for learning paths.

### CODE REVIEW

**Review by Jan-Thurner — APPROVED** [review]
demonstrated in a TestingSession. Works as expected.

**Review by rstief — DISMISSED** [review]
demonstrated in testing session, no error was shown :)

**Review by JohannesWt — DISMISSED** [review]
Code looks good

**Review by tobias-lippert — APPROVED** [review]
I know that we discussed this last year, when you had implemented it😅 Great that it follows the standard REST conventions now. Code LGTM

**Review by Strohgelaender — APPROVED** [review]
code lgtm

**Comment by MaximilianAnzinger** [comment]
> I know that we discussed this last year, when you had implemented it😅 Great that it follows the standard REST conventions now. Code LGTM

@tobias-lippert We discussed it once again and still decided it's an "optional" improvement. I just took the 15 min on the weekend to get it done with 😂

File: src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts [hunk 1]
```diff
@@ -57,8 +57,16 @@ export class LearningPathContainerComponent implements OnInit {
...
             });
         }
-        this.learningPathService.getLearningPathId(this.courseId).subscribe((learningPathIdResponse) => {
-            this.learningPathId = learningPathIdResponse.body!;
+        this.learningPathService.getLearningPathId(this.courseId).subscribe({
+            next: (learningPathIdResponse) => {
+                this.learningPathId = learningPathIdResponse.body!;
+            },
+            error: () => {
+                // if the learning path does not exist, we do need to create it
```
Comment (krusche) [inline src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts:65]: Does it really make sense to generate the learning path for every error that could occur? I think the code would be more robust if you check that the server returned an appropriate error message, e.g. a 404

**Review by krusche — CHANGES_REQUESTED** [review]
see my inline comment

**Review by Strohgelaender — APPROVED** [review]

**Review by Jan-Thurner — APPROVED** [review]
new errornhandling lgtm

**Review by JohannesWt — APPROVED** [review]

**Review by tobias-lippert — APPROVED** [review]

**Pull request merged**